internal/worker/api/permission_test.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
```

Comment:
These tests don't really affirm that authorization is working from the top level. Could you add short authorization tests for the gRPC endpoints themselves instead?

Reply:
Yes, sure.

Reply:
@Joerger can you check if the integration tests are fine?